frontend/src/reducers/user.js

```diff
@@ -0,0 +1,95 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+const initialState = {
+    login: {
+        accessToken: null,
+        userId: 0,
+        secretMessage: null,
+        errorMessage: null,
+    },
+  }
+
+export const user = createSlice({
+    name: 'user',
+    initialState: initialState,
+    reducers: {
+        setAccessToken: (state, action) => {
+            const { accessToken } = action.payload
+            state.login.accessToken = accessToken
+        },
+        setUserId: (state, action) => {
+            const { userId } = action.payload
+            state.login.userId = userId
+        },
+        setErrorMessage: (state, action) => {
+            const { errorMessage } = action.payload
+            state.login.errorMessage = errorMessage
+        }, 
+        setSecretMessage: (state, action) => {
+          const { secretMessage } = action.payload
+          state.secretMessage = secretMessage
+        }
+    }
+})
+
+          // Thunks
```

Comment:
We are very impressed that you decided to apply the thunks? methodology! It's a great way to keep API calls together with its redux store f.e